catboost/app/mode_snapshot_to_model.cpp

```diff
@@ -0,0 +1,84 @@
+#include "modes.h"
+#include "bind_options.h"
+
+#include <catboost/libs/algo/learn_context.h>
+#include <catboost/libs/algo/split.h>
+#include <catboost/libs/helpers/progress_helper.h>
+#include <catboost/libs/model/model_build_helper.h>
+#include <library/getopt/small/last_getopt.h>
+
+
+struct TLoadSnapshotParams {
+    TString SnapshotPath;
+    TString ModelPath;
+
+    void BindParserOpts(NLastGetopt::TOpts& parser) {
+        parser.AddLongOption('s', "snapshot-file", "path to load snapshot from")
+                .StoreResult(&SnapshotPath)
+                .RequiredArgument("PATH");
+        parser.AddLongOption('m', "model-file", "path to save model at")
+                .StoreResult(&ModelPath)
+                .RequiredArgument("PATH");
+    }
+};
+
+
+TLearnProgress LoadSnapshot(const TString& snapshotPath) {
+    CB_ENSURE(NFs::Exists(snapshotPath), "Snapshot file doesn't exist: " << snapshotPath);
+    try {
+        TLearnProgress learnProgress;
+        TProgressHelper(ToString(ETaskType::CPU)).CheckedLoad(snapshotPath, [&](TIFStream* in) {
+            TRestorableFastRng64 rand(0);
+            ::Load(in, rand);
+            learnProgress.Load(in, true);
+        });
+        CATBOOST_INFO_LOG << "Snapshot is loaded from: " << snapshotPath << Endl;
+        return learnProgress;
+    } catch (...) {
+        CATBOOST_ERROR_LOG << "Can't load progress from snapshot file: " << snapshotPath << Endl;
+        throw;
+    }
+}
+
+
+TObliviousTrees BuildObliviousTrees(const TLearnProgress& learnProgress) {
+    try {
+        TObliviousTreeBuilder builder(learnProgress.FloatFeatures, learnProgress.CatFeatures,
+                                      learnProgress.ApproxDimension);
+        for (size_t treeId = 0; treeId < learnProgress.TreeStruct.size(); ++treeId) {
+            TVector <TModelSplit> modelSplits;
+            for (const auto &split : learnProgress.TreeStruct[treeId].Splits) {
```

Comment:
Please check out the style guide for things like space after TVector and position of &. Here it should be:
TVector<TModelSplit>
const auto& split